substrate/frame/support/procedural/src/pallet/parse/mod.rs

```diff
@@ -560,10 +563,37 @@ mod keyword {
 	syn::custom_keyword!(composite_enum);
 }
 
+/// The possible values for the `#[pallet::config]` attribute.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+enum ConfigValue {
+	/// `#[pallet::config(with_default)]`
+	WithDefault(keyword::with_default),
+	/// `#[pallet::config(without_metadata)]`
+	WithoutMetadata(keyword::without_metadata),
+}
+
+impl syn::parse::Parse for ConfigValue {
+	fn parse(input: syn::parse::ParseStream) -> syn::Result<Self> {
+		let lookahead = input.lookahead1();
+
+		if lookahead.peek(keyword::with_default) {
+			input.parse().map(ConfigValue::WithDefault)
+		} else if input.peek(keyword::without_metadata) {
+			input.parse().map(ConfigValue::WithoutMetadata)
+		} else {
+			Err(input.error("expected `with_default` or `without_metadata`"))
+		}
```

Comment:
If you want to use lookahead you can simplify like this:

https://docs.rs/syn/latest/syn/parse/struct.Lookahead1.html
```suggestion
		} else if lookahead.peek(keyword::without_metadata) {
			input.parse().map(ConfigValue::WithoutMetadata)
		} else {
			Err(lookahead.error())
		}
```